### PR TITLE
feat: use resultcard in lnurlchannel

### PR DESCRIPTION
**Pull request by im-adithya**
### Describe the changes you have made in this PR

Align the success screen of LNURLChannel

### Link this PR to an issue [optional]

#1379

### Type of change

(Remove other not matching type)

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes [optional]

Screenshot from 2022-10-07 22-34-17

### How has this been tested?

Manually

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Comment by escapedcat** [comment]
image

Should we take care of that `undefined` in the response? Like say "No nodeId available" or just leave it out if it's undefined?

**Comment by im-adithya** [comment]
image

**Review by escapedcat — APPROVED** [review]
ack, tested with fiatjefs playground

File: src/app/screens/LNURLChannel/index.tsx [hunk 1]
```diff
@@ -55,7 +58,10 @@ function LNURLChannel() {
         throw new Error(`Failed to call callback: ${callbackResponse.message}`);
       }
 
-      setSuccessMessage(`${t("success")}. ${details.k1} ${nodeId}`);
+      setSuccessMessage(
+        t("success", { name: origin ? origin.name : details.domain })
+      );
+      setResponse(`${details.k1} ${nodeId ? nodeId : `[no nodeId available]`}`);
```
Comment (reneaaron) [inline src/app/screens/LNURLChannel/index.tsx:64]: Is the nodeId supposed to be null here?

Comment (bumi) [reply src/app/screens/LNURLChannel/index.tsx:64]: when a node id is not available this channel thing does not work. 
so I don't think that this message should ever happen and thus should be needed.

Comment (escapedcat) [reply src/app/screens/LNURLChannel/index.tsx:64]: In this case it might just be related to testing it with fiatjef?

Comment (bumi) [reply src/app/screens/LNURLChannel/index.tsx:64]: so nodeId comes from line 47. where we do the getInfo call. 
we request the information of the user's connected node and get the node ID from that.

that node id is sent to the callback URL - the service send opens a channel with the user's node (that's why the nodeId is needed).

this whole channel thing works only if there is a node ID available. 
it must never be null. so this `setResponse` does not make sense. 


you could add a check above that if nodeId is falsy then it shows an error and does not even try to do the lnurl-channel call.


I can only imagine that you use an Alby account or some custodial account when you run this. That is not your node so you can not open a a channel to it, or rather you would not. 

the getinfo call likely also does not return the nodeid.

**Review by bumi — CHANGES_REQUESTED** [review]

**Comment by escapedcat** [comment]
lgtm, we see the k1 and the node-id
image

File: src/app/screens/LNURLChannel/index.tsx [hunk 2]
```diff
@@ -55,7 +63,10 @@ function LNURLChannel() {
         throw new Error(`Failed to call callback: ${callbackResponse.message}`);
       }
 
-      setSuccessMessage(`${t("success")}. ${details.k1} ${nodeId}`);
+      setSuccessMessage(
+        t("success", { name: origin ? origin.name : details.domain })
+      );
+      setResponse(`${details.k1} ${nodeId}`);
```
Comment (bumi) [inline src/app/screens/LNURLChannel/index.tsx:69]: Those are some internal information I don't thinks this is useful for the user.

Let's only show the success message and no response details.

(it's enough to say "Channel request sent successfully to {{name}}" )

Comment (im-adithya) [reply src/app/screens/LNURLChannel/index.tsx]: Makes sense, done!

**Comment by escapedcat** [comment]
I assume bumi wanted to merge

**Comment by bumi** [comment]
You had approved it. (though a bit early)

**Comment by escapedcat** [comment]
Yeah, checked again today

**Pull request merged**